File: src/ru/sadv1r/afc/ideaPlugin/Achievable.java
package ru.sadv1r.afc.ideaPlugin;

import com.intellij.openapi.components.ServiceManager;

/**
 * Created on 12/10/15.
 *
 * @author sadv1r
 * @version 0.1
 */
public interface Achievable {
    Stats stats = ServiceManager.getService(Stats.class);

    String getName();
    String getText();
    int getProgress();
    int getFinish();
}

File: src/main/kotlin/ru/sadv1r/afc/ideaPlugin/Achievable.kt
package ru.sadv1r.afc.ideaPlugin

import com.intellij.openapi.components.ServiceManager

abstract class Achievable(
    open val name: String,
    open val text: String,
    open val progress: Int = 0,
    open val finish: Int
){
    companion object{
        val stats: Stats = ServiceManager.getService(Stats::class.java)
    }
}